homeassistant/components/device_tracker/traccar.py

```diff
@@ -0,0 +1,117 @@
+"""
+Support for Traccar location sharing.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.traccar/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, SOURCE_TYPE_GPS)
+from homeassistant.const import ATTR_ID, CONF_PASSWORD, \
+    CONF_USERNAME, CONF_URL
+import homeassistant.util.dt as dt
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.helpers.typing import ConfigType
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ADDRESS = 'address'
+ATTR_FULL_NAME = 'full_name'
+ATTR_LAST_SEEN = 'last_seen'
+ATTR_SPEED = 'speed'
+ATTR_ACCURACY = 'accuracy'
+ATTR_BATTERY = 'battery_level'
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.url,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_scanner(hass, config: ConfigType, see, discovery_info=None):
+    """Set up the scanner."""
+    scanner = TraccarScanner(hass, config, see)
+    return scanner.success_init
+
+
+class TraccarScanner(object):
+
+    def __init__(self, hass, config: ConfigType, see) -> None:
+        """Initialize the scanner."""
+
+        self.see = see
+        self.url = config.get(CONF_URL)
+        self.username = config.get(CONF_USERNAME)
+        self.password = config.get(CONF_PASSWORD)
```

Comment:
You should check the credentials during the setup of the platform and make the setup fails.